### DISCUSSION TITLE
How to make state slice more sensitive to changes

### DISCUSSION THREAD

**Discussion by haibert** [pmndrs/zustand#883] [Q&A]
Thank you for this amazing library.

Lest say I have an array stored in zustand and im reaching for it from a component like so

``` javascript
  const array = useContent((state) => state.myArray))
```

When I make changes to a property that is stored inside an object inside of this array it does not cause any re-renders which might be a good out of the box behavior for most cases. but what if I want it to be super sensitive and cause a re-render even if just a prop inside of an object inside the array is changed?

**Comment by dai-shi** [comment]
While it's not recommended, because it doesn't work well with react concurrent rendering and/or memoization, you could add a version number or something.

```js
const useStore = create((set) => ({
  version: 0, // update the number (immutably) to force update.
  // ...
}))

  // do not use a selector
  const { myArray: array } = useStore()
```
